### PR TITLE
Fix GetTransactions query behaviour

### PR DESCRIPTION
**Pull request by Akvinikym**
[IR-135](https://jira.hyperledger.org/browse/IR-135)

### Description of the Change

GetTransactions query returned list of only those transactions, which hashes were correct and the user had a permission to view them. This behaviour was considered incorrect, so now this query is going to return a stateful error, if at least one of the hashes does not meet the described requirement.

### Benefits

Correct behaviour of the query.

### Possible Drawbacks 

Less usability?

### CODE REVIEW

File: test/module/irohad/ametsuchi/postgres_query_executor_test.cpp [hunk 1]
```diff
@@ -1468,31 +1468,27 @@ namespace iroha {
...
      */
-    TEST_F(GetTransactionsHashExecutorTest, ValidMyAccount) {
-      addPerms({shared_model::interface::permissions::Role::kGetMyTxs});
+    TEST_F(GetTransactionsHashExecutorTest, InvalidBadHash) {
+      addPerms({shared_model::interface::permissions::Role::kGetAllTxs});
 
       commitBlocks();
 
-      std::vector<decltype(hash1)> hashes;
+      std::vector<decltype(hash3)> hashes;
```
Comment (kamilsa) [inline test/module/irohad/ametsuchi/postgres_query_executor_test.cpp]: Why to change that?

File: irohad/ametsuchi/impl/postgres_query_executor.cpp [hunk 2]
```diff
@@ -619,6 +620,14 @@ namespace iroha {
             return (sql_.prepare << cmd, soci::use(creator_id_, "account_id"));
           },
           [&](auto range, auto &my_perm, auto &all_perm) {
+            if (boost::size(range) != q.transactionHashes().size()) {
+              // at least one of the hashes in the query was invalid -
+              // nonexistent or permissions were lacked
+              return this->logAndReturnErrorResponse(
+                  QueryErrorType::kStatefulFailed,
+                  "At least one of the supplied hashes is incorrect",
+                  4);
```
Comment (luckychess) [inline irohad/ametsuchi/impl/postgres_query_executor.cpp]: Add TODO please about getting rid of magic numbers here.

File: test/integration/acceptance/get_transactions_test.cpp [hunk 3]
```diff
@@ -172,15 +172,17 @@ TEST_F(GetTransactions, InvalidSignatures) {
...
  */
 TEST_F(GetTransactions, NonexistentHash) {
   auto check = [](auto &status) {
     ASSERT_NO_THROW({
       const auto &resp =
-          boost::get<const shared_model::interface::TransactionsResponse &>(
+          boost::get<const shared_model::interface::ErrorQueryResponse &>(
               status.get());
-      ASSERT_EQ(resp.transactions().size(), 0);
+      ASSERT_EQ(resp.errorCode(), 4);
```
Comment (luckychess) [inline test/integration/acceptance/get_transactions_test.cpp]: Same TODO here.

File: test/module/irohad/ametsuchi/postgres_query_executor_test.cpp [hunk 4]
```diff
@@ -1468,31 +1468,27 @@ namespace iroha {
...
                        .build();
       auto result = executeQuery(query);
-      checkSuccessfulResult<shared_model::interface::TransactionsResponse>(
-          std::move(result), [this](const auto &cast_resp) {
-            ASSERT_EQ(cast_resp.transactions().size(), 2);
-            ASSERT_EQ(cast_resp.transactions()[0].hash(), hash1);
-            ASSERT_EQ(cast_resp.transactions()[1].hash(), hash2);
-          });
+      checkStatefulError<shared_model::interface::StatefulFailedErrorResponse>(
+          std::move(result), 4);
```
Comment (luckychess) [inline test/module/irohad/ametsuchi/postgres_query_executor_test.cpp]: And here.

**Pull request merged**